Documentation/contributing/development/contributing_guide.rst

```diff
@@ -238,78 +244,88 @@ Getting a pull request merged
    #. Address any feedback received from the reviewers
    #. You can push individual commits to address feedback and then rebase your
       branch at the end before merging.
+   #. Once you have addressed the feedback, re-request a review from the
+      reviewers that provided feedback. This ensures that the reviewers are
+      notified again that your PR is ready for subsequent review.
 
 #. Owners of the repository will automatically adjust the labels on the pull
    request to track its state and progress towards merging.
 
 #. Once the PR has been reviewed and the CI tests have passed, the PR will be
    merged by one of the repository owners. In case this does not happen, ping
-   us on Slack.
-
-#. If reviewers have requested changes and those changes have been addressed,
-   re-request a review for the reviewers that have requested changes. Otherwise,
-   those reviewers will not be notified and your PR will not receive any
-   reviews. If the PR is considerably large (e.g. with more than 200 lines
-   changed and/or more than 6 commits) create new commit for each review. This
-   will make the review process smoother as GitHub has limitations that
-   prevents reviewers from only seeing the new changes added since the last time
-   they have reviewed a PR. Once all reviews are addressed those commits should
-   be squashed against the commit that introduced those changes. This can be
-   easily accomplished by the usage of ``git rebase -i origin/master`` and in
-   that windows, move these new commits below the commit that introduced the
-   changes and replace the work ``pick`` with ``fixup``. In the following
-   example, commit ``d2cb02265`` will be meld into ``9c62e62d8`` and commit
-   ``146829b59`` will be meld into ``9400fed20``.
-
-       ::
-
-           pick 9c62e62d8 docs: updating contribution guide process
-           fixup d2cb02265 joe + paul + chris changes
-           pick 9400fed20 docs: fixing typo
-           fixup 146829b59 Quetin and Maciej reviews
-
-   Once this is done you can perform push force into your branch and request for
-   your PR to be merged.
+   us on Slack in the #development channel.
+
+Handling large pull requests
+----------------------------
+
+If the PR is considerably large (e.g. with more than 200 lines
+changed and/or more than 6 commits) create new commit for each review. This
+will make the review process smoother as GitHub has limitations that
+prevents reviewers from only seeing the new changes added since the last time
+they have reviewed a PR. Once all reviews are addressed those commits should
+be squashed against the commit that introduced those changes. This can be
+accomplished by the usage of ``git rebase -i upstream/master`` and in
```

Comment:
I recently add `--autosquash` flag, so that fixup commits are automatically moved after underlying commit.

Reply:
It works well, but given that these instructions target users who are not too familiar with GitHub, I'd stick with providing a single solution for doing the task. Just my opinion, though.

Reply:
oh right, it's making sense :100:

Reply:
I use `--autosquash` too :rocket: 

Overall I'm not super convinced by this paragraph - if you're familiar with Git then you can definitely make these instructinos work, or like @sayboras you may even have more advanced workflows. I think there is still a bit more that's unsaid in this paragraph around how you collect changes in your commits and make them friendly to rebase / squash. But also, this is the Cilium documentation and not the git documentation. We can only go so far.

One thing I might additionally consider is just adding a paragraph before this to emphasize breaking changes into smaller commits/PRs with their motivation. That will do even more for making the whole process friendlier for both the reviewer and contributor.

Reply:
> One thing...

That's a good idea. Although I wonder how many contributors go through the details of this doc before opening a PR. Maybe we should add a checkbox “[ ] I read the docs” to the PR template :)

Reply:
Not a bad idea, we could link directly to the relevant docs from the template to help spread the knowledge about these contributor resources.

Reply:
I did end up adding a new commit for `One thing ...`: https://github.com/cilium/cilium/pull/21056/commits/c3a7c76ef746a06b095aefb018c17c7aa649654b

Reply:
Yes, I link the idea of linking to the docs (likely as a comment) from the PR template.
New commit looks good, thanks!